### ISSUE TITLE
stop-iteration-return false positive for next() with default argument

### ISSUE THREAD

**Issue by gcolombeau** [PyCQA/pylint#1830]
### Steps to reproduce
```python
def test_next_default_gen():
    yield next(iter([]), 42)
```

### Current behavior
`Do not raise StopIteration in generator, use return statement instead (stop-iteration-return)`

### Expected behavior
This case should not trigger a refactor message as the ``StopIteration`` is substituted by the default argument value.

### pylint --version output
```
No config file found, using default configuration
pylint 1.8.1, 
astroid 1.6.0
Python 3.6.3 (default, Dec  7 2017, 15:52:01) 
[GCC 5.4.0 20160609]
```

**Comment by brycepg** [comment]
Thanks for the report @gcolombeau !

**Comment by PCManticore** [comment]
Makes sense!
